From c5cfae7598150b1dcbb7bcfbc881479575199a64 Mon Sep 17 00:00:00 2001
From: "Uwe L. Korn" <uwe.korn@quantco.com>
Date: Thu, 23 Nov 2023 09:05:37 +0000
Subject: [PATCH 15/18] Remove some usage of absl::str_format in CUDA

nvcc from CUDA 11 is unable to correctly handle the new C++ features
used in this template. We use a different implementation that should
yield the same results without the usage of absl.
---
 tensorflow/core/framework/resource_base.h | 10 ++++++++--
 tensorflow/core/framework/resource_var.cc |  2 ++
 2 files changed, 10 insertions(+), 2 deletions(-)

diff --git a/tensorflow/core/framework/resource_base.h b/tensorflow/core/framework/resource_base.h
index 46a76ea094c..2f8d0a21878 100644
--- a/tensorflow/core/framework/resource_base.h
+++ b/tensorflow/core/framework/resource_base.h
@@ -18,8 +18,9 @@ limitations under the License.
 
 #include <cstdint>
 #include <string>
+#include <sstream>
 
-#include "absl/strings/str_format.h"
+// #include "absl/strings/str_format.h"
 #include "tensorflow/core/lib/core/refcount.h"
 #include "tensorflow/core/lib/core/status.h"
 #include "tensorflow/core/platform/errors.h"
@@ -41,7 +42,12 @@ class ResourceBase : public core::WeakRefCounted {
 
   // Returns a name for ref-counting handles.
   virtual std::string MakeRefCountingHandleName(int64_t resource_id) const {
-    return absl::StrFormat("Resource-%d-at-%p", resource_id, this);
+    std::ostringstream stringStream;
+    stringStream << "Resource-";
+    stringStream << resource_id;
+    stringStream << "-";
+    stringStream << reinterpret_cast<uint64_t>(this);
+    return stringStream.str();
   }
 
   // Returns memory used by this resource.
diff --git a/tensorflow/core/framework/resource_var.cc b/tensorflow/core/framework/resource_var.cc
index 0f5ed5560d6..b1adf7180a7 100644
--- a/tensorflow/core/framework/resource_var.cc
+++ b/tensorflow/core/framework/resource_var.cc
@@ -18,6 +18,8 @@ limitations under the License.
 #include "tensorflow/core/framework/resource_handle.h"
 #include "tensorflow/core/graph/graph_def_builder.h"
 
+#include "absl/strings/str_format.h"
+
 namespace tensorflow {
 
 Status Var::AsGraphDef(GraphDefBuilder* builder, Node** out) const {
